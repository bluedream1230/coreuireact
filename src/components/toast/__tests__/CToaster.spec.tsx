import * as React from 'react'
import { render /* ,screen */ /* ,fireEvent */ /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CToast, CToaster } from '../../../index'

test('loads and displays CToaster component', async () => {
  const { container } = render(<CToaster>Test</CToaster>)
  expect(container).toMatchSnapshot()
})

test('CToaster customize', async () => {
  const { container } = render(
    <CToaster className="bazinga" placement="bottom-start">
      <CToast autohide={false}>Toast</CToast>
    </CToaster>,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveClass('toaster')
  expect(container.firstChild).toHaveClass('toast-container')
  expect(container.firstChild).toHaveClass('p-3')
  expect(container.firstChild).toHaveClass('bottom-0')
  expect(container.firstChild).toHaveClass('start-0')
})
