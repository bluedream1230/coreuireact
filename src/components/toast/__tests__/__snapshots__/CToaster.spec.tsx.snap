// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CToaster customize 1`] = `<div />`;

exports[`loads and displays CToaster component 1`] = `
<div>
  <div
    class="toaster toast-container p-3"
  >
    Test
  </div>
</div>
`;
